tests/src/Functional/PublicationTest.php

```diff
@@ -0,0 +1,151 @@
+use Drupal\Tests\TestFileCreationTrait;
+
+/**
+ * Tests the Publication content type.
+ *
+ * @group oe_starter_content
+ */
+class PublicationTest extends BrowserTestBase {
+
+  use MediaTypeCreationTrait;
+  use TestFileCreationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'oe_starter_content_publication',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $defaultTheme = 'stark';
+
+  /**
+   * Tests the creation of Publication content through the UI.
+   */
+  public function testCreatePublication(): void {
+    $assert_session = $this->assertSession();
+    $page = $this->getSession()->getPage();
+
+    // Create an image entity to be embedded.
+    $image_file = File::create([
+      'uri' => $this->getTestFiles('image')[0]->uri,
+    ]);
+    $image_file->save();
+    $image_media = Media::create([
+      'bundle' => 'image',
+      'name' => 'Example image',
+      'oe_media_image' => [
+        [
+          'target_id' => $image_file->id(),
+          'alt' => 'Image alt',
+          'title' => 'Image title',
+        ],
+      ],
+    ]);
+    $image_media->save();
+
+    // Create a document entity to be embedded.
+    $document_file = File::create([
+      'uri' => $this->getTestFiles('text')[0]->uri,
+    ]);
+    $document_file->save();
+    $document_media = Media::create([
+      'bundle' => 'document',
+      'oe_media_file_type' => 'local',
+      'oe_media_file' => [
+        'target_id' => (int) $document_file->id(),
+      ],
+      'name' => 'Example document',
+    ]);
+    $document_media->save();
+
+    // Login with permission to create publication content.
+    $user = $this->drupalCreateUser([
+      'create oe_sc_publication content',
+      'view media',
+    ]);
+    $this->drupalLogin($user);
+
+    // Attempt to create a publication with no values, to test required fields.
+    $this->drupalGet('node/add/oe_sc_publication');
+    $page->pressButton('Save');
+    // The node is not saved.
+    $assert_session->elementTextEquals('css', 'h1', 'Create Publication');
+    $assert_session->elementsCount('css', 'input.error', 2);
+    $assert_session->pageTextContains('Title field is required.');
+    $assert_session->pageTextContains('Use existing media field is required.');
+
+    // Create a publication with minimal required values.
+    $this->drupalGet('node/add/oe_sc_publication');
+    $page->fillField('Title', 'Publication page');
+    $page->fillField(
+      'Use existing media',
+      $document_media->label() . ' (' . $document_media->id() . ')',
+    );
+    $page->pressButton('Save');
+
+    // Assert only the required fields.
+    $assert_session->elementTextEquals('css', 'h1', 'Publication page');
+    $publication_date = $page->find('css', 'time');
+    $publication_date->hasAttribute('datetime');
+    $this->assertMatchesRegularExpression("/^[0-9]{4}-(0[1-9]|1[0-2])-(0[1-9]|[1-2][0-9]|3[0-1])$/", $publication_date->getText());
```

Comment:
I know why we are using regex here, but honestly I dont like it, better for me would be 
`$this->assertSame(date('Y-m-d'), $publication_date->getText());`
or even better than that use 
`drupal/datetime_testing`
I'll have to discuss this in the team

Reply:
Ok, so conclusion, use https://www.drupal.org/project/datetime_testing, this way you can set a static date and check for it, see how it was done here https://github.com/openeuropa/oe_theme/blob/4d2c9e54a2f22d9d2c08490dc66786ed40359003/tests/src/Kernel/ConsultationRenderTest.php#L66